enhancements/testing/ephemeral-agent-installer-e2e-tests.md

```diff
@@ -0,0 +1,275 @@
+
+### Goals
+
+This proposal will focus using this new e2e framework in the new ephemeral agent based installer 
+image project only. 
+
+### Non-Goals
+
+* Re-factoring dev-scripts and assisted-test-infra for reasons not related to the new 
+ephemeral agent based installer image project.
+* Getting buy in from other installer teams (Assisted Installer, IPI) to use this new 
+framework in their e2e tests for the upcoming release.
+
+This is not to say that we shouldn't talk to other installer teams about this and 
+collaborate with them. It is just not a goal to have them use the same e2e framework as the 
+new project.
+
+## Proposal
+
+Create an e2e testing framework for the ephemeral agent based installer image.
+
+### User Stories
+
+#### Run Assisted Installer Outside k8s
+
+As a developer, I need to run Assisted Installer without installing a k8s platform first 
+so that I can test the installation of a users first cluster, cluster 0.
+
+This is the overall purpose of this ephemeral agent based installer image project. 
+Unfortunately, neither of the existing frameworks actually can actually assist with this. 
+
+* The Assisted Installer e2e tests that run in dev-scripts first use the IPI installer 
+to install OCP and then the infrastructure-operator on this OCP cluster making it a hub 
+cluster.
+
+* The Assisted Installer e2e tests that run in assisted-test-infra installs minikube first 
+and then runs Assisted Installer on top of it.
+
+Which means in both existing e2e frameworks we first setup some form of k8s first. Given
+the goal of the ephemeral agent based installer project to install a users first cluster, 
+cluster 0, this is highly undesirable.
+
+While there is a flow to run Assisted Installer in podman this is a manually supported 
+flow. The support of it in the e2e frameworks is mostly abandoned by the Assisted Installer 
+team in favor of infrastructure-operator and the hosted SaaS solution.
+
+#### Setup Environment for Install Without Installing OCP or k8s
+
+As a developer, I need to set up VMs so that I can simulate the installation of a users 
+first cluster using the ephemeral based agent installer image.
+
+Both existing frameworks today set up VMs using libvirt today to test a users OCP install.
+However, they also both immediately use those VMs to start the installation process. There
+isn't an easy way to stop the current frameworks from using IPI and Assisted Installer
+after the VM creation.
+
+
+#### Launch a VM with the Ephemeral Agent Based Installer Image
+
+As a developer, I need to launch a VM with the built ephemeral agent based installer
+image so that I can test and validate this image.
+
+Both existing frameworks don't have an easy way to do this. Which makes sense because
+they predate the ephemeral agent based installer project. There also isn't an easy way
+to use an existing routine to just launch a generic ISO image which would have been
+something we could have re-purposed. 
+
```

Comment:
https://gist.github.com/beekhof/35137d9f8e93b4512d64e734110cd4aa is an example of another option not already considered here. For the agent-based installer, there's no particular reason to create the VMs before you're ready to boot them from the ISO.

Reply:
That would work, but FWIW the reason I'm suggesting reuse of the existing metal3-dev-env roles is those include support for more complex scenarios, e.g multiple disks, configurable support for multiple networks, enable/disable DHCP, etc - it's not just how to boot a VM with an ISO.